pkg/infrastructure/vsphere/clusterapi/import.go

```diff
@@ -58,9 +59,17 @@ func checkOvaSecureBoot(ovfEnvelope *ovf.Envelope) bool {
 }
 
 func importRhcosOva(ctx context.Context, session *session.Session, folder *object.Folder, cachedImage, clusterID, tagID, diskProvisioningType string, failureDomain vsphere.FailureDomain) error {
-	name := fmt.Sprintf("%s-rhcos-%s-%s", clusterID, failureDomain.Region, failureDomain.Zone)
+	// Name originally was cluster id + fd.region + fd.zone.  This could cause length of ova to be longer than max allowed.
+	// So for now, we are going to make cluster id  + fd.name
+	name := utils.GenerateVSphereTemplateName(clusterID, failureDomain.Name)
 	logrus.Infof("Importing OVA %v into failure domain %v.", name, failureDomain.Name)
 
+	// OVA name must not exceed 80 characters
+	if len(name) > 80 {
+		logrus.Warningf("Unable to generate ova template name due to exceeding 80 characters. Cluster=\"%v\" Failure Domain=\"%v\" results in \"%v\"", clusterID, failureDomain.Name, name)
+		return fmt.Errorf("ova name \"%v\" exceeed 80 characters (%d)", name, len(name))
```

Comment:
Not a vSphere expert here, but wondering if we can just truncate template name to 80 characters instead of erroring out. And also we should be checking the length of the generated template name within GenerateVSphereTemplateName() and taking corrective action (like truncation) or erroring out from there.

Reply:
If we truncate and end up with template names for two failure domains with the same name, we will have an error back from vsphere (filename needs to be unique).  I can add the logic to truncate, and then do an additional check to make sure none of the generated names match, but we will still have a case where an error will occur.  The thought here was if its too long based on the generated name, the customer can change configuration for each failure domain to have a shorter FD name, or they can upload the template with the name they want and then set the template name in each FD.

I can totally move the check into the generate function.  I'll try that out and make that changes.  Thanks.

Reply:
+1 to @vr4manta 's reasoning here.  name conflicts are quite possible if we truncate the name.